clang/tools/libclang/CDependencies.cpp

```diff
@@ -179,7 +185,9 @@ static CXFileDependencies *getFullDependencies(
       for (const ModuleID &MID : MD.ClangModuleDeps)
         Modules.push_back(MID.ModuleName + ":" + MID.ContextHash);
       M.ModuleDeps = cxstring::createSet(Modules);
-      M.BuildArguments = cxstring::createSet(MD.getNonPathCommandLine());
+      M.BuildArguments =
+          cxstring::createSet(ModDepCmd == Full ? MD.getFullCommandLine()
+                                                : MD.getNonPathCommandLine());
```

Comment:
The APIs called here are a bit confusing. IIUC, neither API returns the full command-line, and both skip the "paths". It'd be nice to clean up the naming (outside of this commit of course as it's not really related, I'm just noticing now; doesn't need to happen first).

Reply:
That's a good point.

What I've come up for `ModuleDeps` is this:
* `getCanonicalCommandLine(LookupPCMPath, LookupModuleDeps)`
  * Returns command-line generated from `CompilerInvocation` (hence "canonical"), with path arguments.
* `getCanonicalCommandLineWithoutModulePaths()`
  * Returns the same, but without path arguments.
* `getAdditionalArgsWithoutModulePaths()`
  * Returns additional arguments starting with `-remove-preceeding-explicit-module-build-incompatible-options`.

Consistent with `ModuleDeps` naming, the `FullDependencies` have:
* `getAdditionalArgs(LookupPCMPath, LookupModuleDeps)`
  * Returns additional arguments (e.g. `-fno-implicit-modules`) with path arguments.
* `getAdditionalArgsWithoutModulePaths()`
  * Returns the same but without path arguments.

I've added you as a reviewer to [D101051](https://reviews.llvm.org/D101051), we can move the discussion there.